store/store.go

```diff
@@ -51,44 +50,41 @@ type Store struct {
 
 // NewStore creates a new EDS Store under the given basepath and datastore.
 func NewStore(params *Parameters, basePath string) (*Store, error) {
-	if err := params.Validate(); err != nil {
+	err := params.Validate()
+	if err != nil {
 		return nil, err
 	}
 
-	// Ensure the blocks folder exists or is created.
-	blocksFolderPath := filepath.Join(basePath, blocksPath)
-	if err := ensureFolder(blocksFolderPath); err != nil {
-		log.Errorf("Failed to ensure the existence of the blocks folder at '%s': %s", blocksFolderPath, err)
-		return nil, fmt.Errorf("ensure blocks folder '%s': %w", blocksFolderPath, err)
-	}
-
-	// Ensure the heights folder exists or is created.
-	heightsFolderPath := filepath.Join(basePath, heightsPath)
-	if err := ensureFolder(heightsFolderPath); err != nil {
-		log.Errorf("Failed to ensure the existence of the heights folder at '%s': %s", heightsFolderPath, err)
-		return nil, fmt.Errorf("ensure heights folder '%s': %w", heightsFolderPath, err)
+	// ensure the blocks dir exists
+	blocksDir := filepath.Join(basePath, blocksPath)
+	if err := mkdir(blocksDir); err != nil {
+		return nil, fmt.Errorf("ensuring blocks directory: %w", err)
```

Comment:
ensureFolder feel closer than `mkdir` to what actually does: create only when missing. And it is similar to other functions like `ensureEmptyFile` and `ensureHeighLing`

Reply:
kk